lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -257,12 +257,24 @@ def get_code_string(self):
 """
         return result
 
+    def _write_query_plan_to_trace_dir(self, plan: LogicalPlan, query_id: str):
+        assert self.trace_dir is not None, "Writing query_plan requires trace_dir to be set"
+        path = os.path.join(self.trace_dir, f"{query_id}/metadata/")
```

Comment:
Since we're using `os.path.join` already please do

`path = os.path.join(self.trace_dir, query_id, "metadata")`

instead.